packages/astro/src/vite-plugin-scanner/index.ts

```diff
@@ -43,12 +44,13 @@ export default function astroScannerPlugin({
 			if (typeof pageOptions.prerender === 'undefined') {
 				pageOptions.prerender = defaultPrerender;
 			}
-
 			// `getStaticPaths` warning is just a string check, should be good enough for most cases
 			if (
 				!pageOptions.prerender &&
 				isServerLikeOutput(settings.config) &&
-				code.includes('getStaticPaths')
+				code.includes('getStaticPaths') &&
+				// this should only be valid for `.astro` files
+				extname(filename) === '.astro'
```

Comment:
Actually I just remembered this could happen for endpoints too, so we need to extend the extensions to `.js` and `.ts` too?

Reply:
If we include `.js` and `.ts` files, we are at square one 🤣

Reply:
Probably good enough to fix the issue since we skip `.md` and `.mdx`, but true it's not so robust 😄